include/utils/str.h

```diff
@@ -119,30 +124,40 @@ void str_append_c(str *dst, const char *src);
  */
 void str_append_buf(str *dst, const char *buf, size_t len);
 
+/**
+ * @brief Replace the string object data with a copy of the C string (null terminated).
+ */
+void str_copy_c(str *dst, const char *src);
+
 /**
  * @brief Replace content in string with something else.
  * @details Replace occurrences of the search string with the replacement string.
  *          Limit parameter can be used to limit replacement to a certain number
- *          of occurences if so wanted (or -1 for unlimited).
+ *          of occurrences if so wanted (or -1 for unlimited).
  * @param dst Target string to modify
  * @param seek String to search and replace
  * @param replacement The replacement value
  * @param limit Number of replacements performed (-1 for unlimited).
  */
 void str_replace(str *dst, const char *seek, const char *replacement, int limit);
 
+/**
+ * @brief Get and remove last character of the string.
+ */
+char str_pop_ch(str *src);
```

Comment:
Afaiks the need for this got dropped as there is now the rstrip function. Is there some specific need for this ?